src/server/report/report.controller.ts

```diff
@@ -56,16 +57,25 @@ export class ReportController {
   })
   @Get('/courses')
   public async getCoursesReport(
-    @Res() res: Response
+    @Res() res: Response,
+      @Query('startYear') start?: string,
+      @Query('endYear') end?: string
```

Comment:
Aha! I was looking for the way to do that in the Nest docs but got hung up on the Validation stuff and didn't think it would work for query params. That's much better. 

I implemented that in 47e7e3aa, though doing so means that the values can no longer be optional; we could use the `DefaultValuePipe`, but that's also tricky when we're computing the default based on the contents of our database. Regardless, there isn't really a use case for having those values be optional anymore, so it's easy enough to just drop that use case from the test suite.